### ISSUE TITLE
Wrong diagnostic: `main` function not found in crate

### ISSUE THREAD

**Issue by dev-ardi** [rust-lang/rust#124935]
Tested on nightly and stable.
```rs
;
fn main() { }
```
```
error: expected item, found `;`
 --> src/main.rs:1:1
  |
1 | ;
  | ^ help: remove this semicolon

error[E0601]: `main` function not found in crate `tests`
 --> src/main.rs:1:2
  |
1 | ;
  |  ^ consider adding a `main` function to `src/main.rs`

For more information about this error, try `rustc --explain E0601`.
error: could not compile `tests` (bin "tests") due to 2 previous errors
```
The second diagnostic is strange.

**Comment by dev-ardi** [comment]
@saethlin any idea why rustbot ignored me?

**Comment by saethlin** [comment]
Just guessing, the missing + for S-has-mcve maybe?

**Comment by fmease** [comment]
Oops, this should've been closed.
